### DISCUSSION TITLE
Install of v2.2.0 fails (not pkg-conf issue)

### DISCUSSION THREAD

**Discussion by GamerBene19** [PyMySQL/mysqlclient#647] [Issue report] [answered]
### Describe the bug

When installing mysqlclient (v.2.2.0) **with pkg-config already installed** the install fails with 
```
root@paperless:/opt/paperless-ngx# python3 -m pip install mysqlclient==2.2.0
Collecting mysqlclient==2.2.0
  Using cached mysqlclient-2.2.0.tar.gz (89 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
  WARNING: Generating metadata for package mysqlclient produced metadata for project name unknown. Fix your #egg=mysqlclient fragments.
Discarding https://files.pythonhosted.org/packages/de/9c/b176826e8994551ce826404dab97e305a4bb76c8b0a4e016fabda2901c71/mysqlclient-2.2.0.tar.gz#sha256=04368445f9c487d8abb7a878e3d23e923e6072c04a6c320f9e0dc8a82efba14e (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.8): Requested unknown from https://files.pythonhosted.org/packages/de/9c/b176826e8994551ce826404dab97e305a4bb76c8b0a4e016fabda2901c71/mysqlclient-2.2.0.tar.gz#sha256=04368445f9c487d8abb7a878e3d23e923e6072c04a6c320f9e0dc8a82efba14e has inconsistent name: filename has 'mysqlclient', but metadata has 'unknown'
ERROR: Could not find a version that satisfies the requirement mysqlclient==2.2.0 (from versions: 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.3.9, 1.3.10, 1.3.11rc1, 1.3.11, 1.3.12, 1.3.13, 1.3.14, 1.4.0rc1, 1.4.0rc2, 1.4.0rc3, 1.4.0, 1.4.1, 1.4.2, 1.4.2.post1, 1.4.3, 1.4.4, 1.4.5, 1.4.6, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.1.0rc1, 2.1.0, 2.1.1, 2.2.0rc1, 2.2.0)
ERROR: No matching distribution found for mysqlclient==2.2.0
```
From what I can tell this is because pip is more strict (see [here](https://stackoverflow.com/a/68126941)) and installing works when using `python3 -m pip install --use-deprecated=legacy-resolverm ysqlclient==2.2.0`

### Environment

```
root@paperless:/opt/paperless-ngx# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy
root@paperless:/opt/paperless-ngx# python3 --version
Python 3.10.12
root@paperless:/opt/paperless-ngx# python3 -m pip --version
pip 22.0.2 from /usr/lib/python3/dist-packages/pip (python 3.10)
```

### How did you install libmysqlclient libraries?

apt-get

### What version of mysqlclient do you use?

(trying to use) 2.2.0

### Docker command to start MySQL server

```bash
none
```

### Minimum but complete code to reproduce

```python
none
```

### Schema and initial data required to reproduce.

```sql
none
```

### Commands, and any other step required to reproduce your issue.

1. Set up fresh Ubuntu 22.04
2. `apt update && apt upgrade`
3. `apt install python3-dev default-libmysqlclient-dev build-essential pkg-config` (as per README.md)
4. Try to install mysqlclient: `python3 -m pip install mysqlclient==2.2.0`

**Comment by methane** [comment, marked as answer]
Use venv instead of sudo pip install.

**Reply by methane** [reply]
When I run your commands, I see this warning.

> [!WARNING]  
> WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

This is why your environment doesn't work.

* mysqlclient uses pyproject.toml and setuptools as a build backend (PEP 517).
* But your system has old setuptools without build backend support.

**Reply by GamerBene19** [reply]
Thanks for the reply and explanation and sorry for the noob mistake  ^^
